src/libcore/sync/atomic.rs

```diff
@@ -84,28 +84,8 @@ use intrinsics;
 use cell::UnsafeCell;
 use fmt;
 
-/// Save power or switch hyperthreads in a busy-wait spin-loop.
-///
-/// This function is deliberately more primitive than
-/// [`std::thread::yield_now`](../../../std/thread/fn.yield_now.html) and
-/// does not directly yield to the system's scheduler.
-/// In some cases it might be useful to use a combination of both functions.
-/// Careful benchmarking is advised.
-///
-/// On some platforms this function may not do anything at all.
-#[inline]
 #[stable(feature = "spin_loop_hint", since = "1.24.0")]
-pub fn spin_loop_hint() {
-    #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
-    unsafe {
-        asm!("pause" ::: "memory" : "volatile");
-    }
-
-    #[cfg(target_arch = "aarch64")]
-    unsafe {
-        asm!("yield" ::: "memory" : "volatile");
-    }
-}
+pub use hint::spin_loop as spin_loop_hint;
```

Comment:
Any reason not to deprecate this one?

Reply:
Because if the other name is unstable, then you can't actually use the replacement.

Reply:
Ah of course :) I'd already put myself in the mindset that this would be stable.